pkg/backuphelpers/backupvars.go

```diff
@@ -0,0 +1,97 @@
+package backuphelpers
+
+import (
+	"fmt"
+	"k8s.io/klog/v2"
+	"reflect"
+	"strings"
+	"sync"
+
+	backupv1alpha1 "github.com/openshift/api/config/v1alpha1"
+	prune "github.com/openshift/cluster-etcd-operator/pkg/cmd/prune-backups"
+)
+
+type Enqueueable interface {
+	Enqueue()
+}
+
+type BackupVar interface {
+	AddListener(listener Enqueueable)
+	SetBackupSpec(spec backupv1alpha1.EtcdBackupSpec)
+	ArgString() string
+}
+
+type BackupConfig struct {
+	enabled   bool
+	spec      backupv1alpha1.EtcdBackupSpec
+	listeners []Enqueueable
+	mux       sync.Mutex
+}
+
+func NewDisabledBackupConfig() *BackupConfig {
+	klog.Infof("@Mustafa : BackupConfig - construct ")
+	return &BackupConfig{
+		enabled: false,
+		mux:     sync.Mutex{},
+	}
+}
+
+func (b *BackupConfig) SetBackupSpec(spec backupv1alpha1.EtcdBackupSpec) {
+	b.mux.Lock()
+	defer b.mux.Unlock()
+
+	klog.Infof("@Mustafa : BackupConfig - Spec %v", spec)
+	if reflect.DeepEqual(b.spec, spec) {
+		return
+	}
+
+	if isEmptyEtcdBackupSpec(spec) {
+		b.enabled = false
+	} else {
+		b.enabled = true
+	}
```

Comment:
this can be simplified
```suggestion
        b.enabled = !isEmptyEtcdBackupSpec(spec)
```

maybe make `spec` a pointer and check for nil instead? 

but I'm wondering, is it empty? because down below it's defined as:
```
reflect.DeepEqual(backupv1alpha1.EtcdBackupSpec{}, spec)
```

Reply:
I need to discuss this, and  also other opinions 

So the idea is to enable the sidecar if there are no `Backup` CRs applied on the cluster by the user

Once the user apply a CR, then this container should be disabled 

Also we need to avoid enabling/disabling randomly, since re-deploying static pods results in a period of cluster instability 

wdyt ? @hasbro17 @dusk125